x-pack/test/api_integration/deployment_agnostic/apis/observability/apm/alerts/helpers/alerting_api_helper.ts

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Client, errors } from '@elastic/elasticsearch';
+import { ParsedTechnicalFields } from '@kbn/rule-registry-plugin/common';
+import pRetry from 'p-retry';
+import { ApmRuleType } from '@kbn/rule-data-utils';
+import { ApmRuleParamsType } from '@kbn/apm-plugin/common/rules/schema';
+import { ApmDocumentType } from '@kbn/apm-plugin/common/document_type';
+import { RollupInterval } from '@kbn/apm-plugin/common/rollup';
+import { ObservabilityApmAlert } from '@kbn/alerts-as-data-utils';
+import type { ApmApiClient } from '../../../../../services/apm_api';
+import type { SupertestWithRoleScopeType } from '../../../../../services';
+
+export const APM_ALERTS_INDEX = '.alerts-observability.apm.alerts-*';
+export const APM_ACTION_VARIABLE_INDEX = 'apm-index-connector-test';
+
+export async function createApmRule<T extends ApmRuleType>({
```

Comment:
Can we move the helper so it can be shared both in the agnostic and apm api tests? Right now we'll have a dublicate

Reply:
we need to duplicate them, both use a different instance of `supertest`

Reply:
@kpatticha  fyi: I've replaced these helpers with `AlertingAPIProvider` [14dc86c](https://github.com/elastic/kibana/pull/199097/commits/14dc86c63272013f6a539b5449d3ebd3b224ffbf)